addon/model.js

```diff
@@ -24,7 +24,7 @@ const FakeAttributeMeta = { isAttribute: true };
 class EmbeddedSnapshot {
   constructor(record) {
     this.record = record;
-    this.modelName = this.record._internalModel.modelName;
+    this.modelName = this.record.modelName;
```

Comment:
shouldn't this either be `this.modelName = this.record._modelName;` or M3._modelName be renamed to M3.modelName ?

Reply:
Good catch! Thinking about it, the change was supposed to not use `_internalModel`, because it will refer to the base model, but `EmbeddedSnapshot` is only used for nested models, which doesn't have `_baseModel`, therefore the line should work as it is. I will revert it.

Reply:
should add a test for this